make-release.sh

```diff
@@ -27,132 +67,185 @@ Usage: $0 --version [VERSION TO RELEASE]
-  # prod-ver should never go down, only up
-  OLDVERSION=$(cat $playbookfile | yq -r '.asciidoc.attributes."prod-ver-patch"') # existing prod-ver-patch version 7.yy.z
-  VERSIONS="${OLDVERSION} ${NEWVERSION}"
-  VERSIONS_SORTED="$(echo $VERSIONS | tr " " "\n" | sort -V | tr "\n" " ")"
+gitTag() {
+  if [[ $TAG_RELEASE -eq 1 ]]; then
+    echo "[INFO] Creating release tag"
+    git checkout "${TARGET_BRANCH}"
+    git pull origin "${TARGET_BRANCH}" || true
+    git tag "${VERSION}"
+    git push origin "${VERSION}" || true
+  fi
+}
+
+versionFormatIsValid() {
+# Validating version format
+[[ $VERSION =~ ^([0-9]+)\.([0-9]+)\.([0-9]+) ]] && MAJOR="${BASH_REMATCH[1]}" ; MINOR="${BASH_REMATCH[2]}"; PATCH="${BASH_REMATCH[3]}"
+case $MAJOR in
+ "") echo "Version $VERSION is not in form MAJOR.MINOR.PATCH"
+     exit 1
+     ;;
+esac
+case $MINOR in
+ "") echo "Version $VERSION is not in form MAJOR.MINOR.PATCH"
+     exit 1
+     ;;
+esac
+case $PATCH in
+ "") echo "Version $VERSION is not in form MAJOR.MINOR.PATCH"
+     exit 1
+     ;;
+esac
+echo "[INFO] Version format for: ${VERSION} is in form MAJOR.MINOR.PATCH."
+}
+
+versionIsIncremented() {
+  # Validation version is incremented, never decremented
+  OLDVERSION=$(yq -r '.asciidoc.attributes."prod-ver-patch"' "$yamlfile") # existing prod-ver-patch version 7.yy.z
+  VERSIONS="${OLDVERSION} ${VERSION}"
+  VERSIONS_SORTED="$(echo "$VERSIONS" | tr " " "\n" | sort -V | tr "\n" " ")"
   # echo "Compare '$VERSIONS_SORTED' with '$VERSIONS '"
-  if [[ "${VERSIONS_SORTED}" != "${VERSIONS} " ]] || [[ "${OLDVERSION}" == "${NEWVERSION}" ]]; then # note trailing space after VERSIONS is required!
-    echo "[WARN] Existing prod-ver ${OLDVERSION} is greater or equal than planned update to ${NEWVERSION}. Version should not go backwards, so nothing to do!"
+  if [[ "${VERSIONS_SORTED}" != "${VERSIONS} " ]] || [[ "${OLDVERSION}" == "${VERSION}" ]]; then # note trailing space after VERSIONS is required!
+    echo "[WARN] Existing prod-ver ${OLDVERSION} is greater or equal than planned update to ${VERSION}. Version should not go backwards, so nothing to do!"
     return 1
-  else 
-    replaceFieldSed $playbookfile 'prod-ver' "\"${BASE1}.${BASE2}\""
-    replaceFieldSed $playbookfile 'prod-ver-patch' "\"$NEWVERSION\""
-    # set prod-prev-ver = 7.y-1
-    (( BASE2=BASE2-1 ))
-    replaceFieldSed $playbookfile 'prod-prev-ver' "\"${BASE1}.${BASE2}\""
   fi
+  echo "[INFO] Updating to version: ${VERSION} is an increment for current version: ${OLDVERSION}."
 }
 
 replaceFieldSed()
 {
   YAMLFILE=$1
   updateName=$2
   updateVal=$3
-  # echo "[INFO] * ${YAMLFILE} ==> ${updateName}: ${updateVal}"
-  sed -i ${YAMLFILE} -r -e "s#(    $updateName: ).+#\1${updateVal}#"
+  echo "[INFO] Updating file: ${YAMLFILE} on branch: ${TARGET_BRANCH}: setting attribute: ${updateName}: ${updateVal}"
+  sed -i "${YAMLFILE}" -r -e "s#(    $updateName: ).+#\1${updateVal}#"
```

Comment:
This could be yq instead of sed but for simply one-line changes I guess this is just as easy :D

Reply:
Interesting comment, I had never thought about using yq to modify YAML.